proxy-node-gateway/src/test/java/uk/gov/ida/notification/resources/HubResponseResourceTest.java

```diff
@@ -130,50 +139,16 @@ public void testsHappyPath() {
         assertEquals(expectedLogOutput, allLogRecords);
     }
 
-    @Test
+    @Test(expected = SessionAttributeException.class)
     public void shouldThrowSamlAttributeErrorIfMissingSessionIsNull() {
-        when(session.getAttribute(SESSION_KEY_SESSION_DATA)).thenReturn(null);
-
-        HubResponseResource resource =  new HubResponseResource(
-            new SamlFormViewBuilder(),
-            translatorProxy
-        );
-
-        assertThatThrownBy(() -> { resource.hubResponse("hub_saml_response", "relay_state", session); })
-            .isInstanceOf(SessionAttributeException.class)
-            .hasMessage("Session data can not be null");
-    }
-
-    @Test
-    public void shouldThrowSamlAttributeErrorIfMissingSessionAttribute() {
-        EidasSamlParserResponse eidasSamlParserResponse = new EidasSamlParserResponse(
-            "eidas_request_id_in_session",
-            "issuer",
-            "",
-            null
-        );
-
-        AuthnRequestResponse vspResponse = new AuthnRequestResponse(
-            "saml-request",
-            "hub_request_id_in_session",
-            UriBuilder.fromUri("http://conector.node").build()
-        );
-
-        GatewaySessionData sessionData = new GatewaySessionData(
-            eidasSamlParserResponse,
-            vspResponse,
-            "eidas_relay_state_in_session"
-        );
-
-        when(session.getAttribute(SESSION_KEY_SESSION_DATA)).thenReturn(sessionData);
+        when(sessionCacheManager.getSession(eq(session.getId()))).thenThrow(SessionAttributeException.class);
 
-        HubResponseResource resource =  new HubResponseResource(
+        HubResponseResource resource = new HubResponseResource(
             new SamlFormViewBuilder(),
-            translatorProxy
+            translatorProxy,
+            sessionCacheManager
         );
 
-        assertThatThrownBy(() -> { resource.hubResponse("hub_saml_response", "relay_state", session); })
-            .isInstanceOf(SessionAttributeException.class)
-            .hasMessage("eidasConnectorPublicKey field may not be empty, eidasDestination field may not be empty");
```

Comment:
Do we not care about the message anymore? I think you shouldn't because you are typing all your exceptions but it is worth noting that you are not checking messages (presumably for that reason)

Reply:
The type of exception being thrown is now different and less likely to be different.